### ISSUE TITLE
Error refreshing while last button is selected

### ISSUE THREAD

**Issue by sw9** [Alexander-Miller/treemacs#8]
Do a treemacs init. Then navigate to the last button, and refresh. I get an error with the following backtrace:
```

Debugger entered--Lisp error: (wrong-type-argument integer-or-marker-p nil)
  get-text-property(nil abs-path)
  button-get(nil abs-path)
  treemacs--prop-at-point(abs-path)
  (let* ((curr-line (line-number-at-pos)) (curr-path (treemacs--prop-at-point (quote abs-path))) (win-start (window-start (get-buffer-window))) (root-btn (treemacs--current-root-btn)) (root (button-get root-btn (quote abs-path)))) (treemacs--build-tree root) (if (and (f-exists\? curr-path) (or treemacs-show-hidden-files (not (s-matches\? treemacs-dotfiles-regex (f-filename curr-path))))) (treemacs--goto-button-at curr-path) (let ((inhibit inhibit-message)) (setq inhibit-message t) (with-no-warnings (goto-line curr-line)) (setq inhibit-message inhibit))) (treemacs-goto-column-1) (set-window-start (get-buffer-window) win-start) (hl-line-mode t) (message "Treemacs buffer refreshed."))
  (progn (select-window (car save-selected-window--state) (quote norecord)) (let* ((curr-line (line-number-at-pos)) (curr-path (treemacs--prop-at-point (quote abs-path))) (win-start (window-start (get-buffer-window))) (root-btn (treemacs--current-root-btn)) (root (button-get root-btn (quote abs-path)))) (treemacs--build-tree root) (if (and (f-exists\? curr-path) (or treemacs-show-hidden-files (not (s-matches\? treemacs-dotfiles-regex (f-filename curr-path))))) (treemacs--goto-button-at curr-path) (let ((inhibit inhibit-message)) (setq inhibit-message t) (with-no-warnings (goto-line curr-line)) (setq inhibit-message inhibit))) (treemacs-goto-column-1) (set-window-start (get-buffer-window) win-start) (hl-line-mode t) (message "Treemacs buffer refreshed.")))
  (unwind-protect (progn (select-window (car save-selected-window--state) (quote norecord)) (let* ((curr-line (line-number-at-pos)) (curr-path (treemacs--prop-at-point (quote abs-path))) (win-start (window-start (get-buffer-window))) (root-btn (treemacs--current-root-btn)) (root (button-get root-btn (quote abs-path)))) (treemacs--build-tree root) (if (and (f-exists\? curr-path) (or treemacs-show-hidden-files (not (s-matches\? treemacs-dotfiles-regex ...)))) (treemacs--goto-button-at curr-path) (let ((inhibit inhibit-message)) (setq inhibit-message t) (with-no-warnings (goto-line curr-line)) (setq inhibit-message inhibit))) (treemacs-goto-column-1) (set-window-start (get-buffer-window) win-start) (hl-line-mode t) (message "Treemacs buffer refreshed."))) (internal--after-with-selected-window save-selected-window--state))
  (save-current-buffer (unwind-protect (progn (select-window (car save-selected-window--state) (quote norecord)) (let* ((curr-line (line-number-at-pos)) (curr-path (treemacs--prop-at-point (quote abs-path))) (win-start (window-start (get-buffer-window))) (root-btn (treemacs--current-root-btn)) (root (button-get root-btn (quote abs-path)))) (treemacs--build-tree root) (if (and (f-exists\? curr-path) (or treemacs-show-hidden-files (not ...))) (treemacs--goto-button-at curr-path) (let ((inhibit inhibit-message)) (setq inhibit-message t) (with-no-warnings (goto-line curr-line)) (setq inhibit-message inhibit))) (treemacs-goto-column-1) (set-window-start (get-buffer-window) win-start) (hl-line-mode t) (message "Treemacs buffer refreshed."))) (internal--after-with-selected-window save-selected-window--state)))
  (let ((save-selected-window--state (internal--before-with-selected-window (get-buffer-window treemacs-buffer)))) (save-current-buffer (unwind-protect (progn (select-window (car save-selected-window--state) (quote norecord)) (let* ((curr-line (line-number-at-pos)) (curr-path (treemacs--prop-at-point ...)) (win-start (window-start ...)) (root-btn (treemacs--current-root-btn)) (root (button-get root-btn ...))) (treemacs--build-tree root) (if (and (f-exists\? curr-path) (or treemacs-show-hidden-files ...)) (treemacs--goto-button-at curr-path) (let (...) (setq inhibit-message t) (with-no-warnings ...) (setq inhibit-message inhibit))) (treemacs-goto-column-1) (set-window-start (get-buffer-window) win-start) (hl-line-mode t) (message "Treemacs buffer refreshed."))) (internal--after-with-selected-window save-selected-window--state))))
  (if treemacs-buffer (let ((save-selected-window--state (internal--before-with-selected-window (get-buffer-window treemacs-buffer)))) (save-current-buffer (unwind-protect (progn (select-window (car save-selected-window--state) (quote norecord)) (let* ((curr-line ...) (curr-path ...) (win-start ...) (root-btn ...) (root ...)) (treemacs--build-tree root) (if (and ... ...) (treemacs--goto-button-at curr-path) (let ... ... ... ...)) (treemacs-goto-column-1) (set-window-start (get-buffer-window) win-start) (hl-line-mode t) (message "Treemacs buffer refreshed."))) (internal--after-with-selected-window save-selected-window--state)))) (message "Treemacs buffer does not exist."))
  (let ((treemacs-buffer (get-buffer treemacs--buffer-name))) (if treemacs-buffer (let ((save-selected-window--state (internal--before-with-selected-window (get-buffer-window treemacs-buffer)))) (save-current-buffer (unwind-protect (progn (select-window (car save-selected-window--state) (quote norecord)) (let* (... ... ... ... ...) (treemacs--build-tree root) (if ... ... ...) (treemacs-goto-column-1) (set-window-start ... win-start) (hl-line-mode t) (message "Treemacs buffer refreshed."))) (internal--after-with-selected-window save-selected-window--state)))) (message "Treemacs buffer does not exist.")))
  treemacs-refresh()
  funcall-interactively(treemacs-refresh)
  call-interactively(treemacs-refresh nil nil)
  command-execute(treemacs-refresh)
```

**Comment by Alexander-Miller** [comment]
Pretty sure I know where this is coming from - fix should be up in a minute.

I also can't reproduce this -  I assume the issue is caused by prop-at-point trying to grab next-button when point is already beyond the last button in the buffer. This shouldn't usually happen and I mostly make sure to keep point in column 1, but this is just to keep the cursor from blinking when it's on top of an icon. That position also needs to change since column 1 is exactly where icons will go when treemacs-indentation is 1.

Long story short: am I right to assume that you've used something other than treemacs-next-line to move point? Because if not we've some debugging to do.

**Comment by sw9** [comment]
Ah that appears to the problem. I only get the error when using end-of-buffer or next-line, not with treemacs-next-line.

**Comment by Alexander-Miller** [comment]
FIx is up. Give it a try and then we can close this issue.

**Comment by sw9** [comment]
Works for me.

**Comment by Alexander-Miller** [comment]
:+1:
